domain/domain.py

```diff
@@ -271,25 +288,34 @@ def get_domain_cell(self, attr, row):
 
         # Remove _nan_ if added due to correlated attributes
         domain.discard('_nan_')
+
         # Add initial value in domain
-        if pd.isnull(row[attr]):
-            domain.update(set(['_nan_']))
-            init_value = '_nan_'
-        else:
-            domain.update(set([row[attr]]))
-            init_value = row[attr]
-        return init_value, list(domain)
+        init_values = ['_nan_']
+        if not pd.isnull(row[attr]):
+            # Assume value in raw dataset is given as ||| separate initial values
+            init_values = row[attr].split('|||')
+        domain.update(set(init_values))
```

Comment:
no need to convert `init_values` to `set` first

Reply:
Good catch.